proxy/http2/Http2Stream.cc

```diff
@@ -691,10 +691,13 @@ Http2Stream::send_response_body(bool call_update)
     // when write_vio is consumed
   } else {
     SCOPED_MUTEX_LOCK(lock, parent->connection_state.mutex, this_ethread());
-    parent->connection_state.send_data_frames(this);
-    this->signal_write_event(call_update);
-    // XXX The call to signal_write_event can destroy/free the Http2Stream.
-    // Don't modify the Http2Stream after calling this method.
+    // send_data_frames returns true if the stream was deleted during processing
+    // Don't do any more processing if the stream has already been deleted
+    if (!parent->connection_state.send_data_frames(this)) {
+      this->signal_write_event(call_update);
```

Comment:
There is a case of the stream is **deleted** but not **freed**, right? In that case, do not calling this function is ok?

Reply:
This particular path (send_data_frames returning true), the stream object (this) was deleted and freed.  So there would be harm in dereferencing this and calling signal_write_event.

Reply:
IIUC, there're no guarantee of "send_data_frames returning true" means "the stream object (this) was deleted and freed".
Because 
- when `Http2ConnectionState::send_data_frames()` calls `Http2ConnectionState::delete_stream()`, it returns true
- `Http2ConnectionState::delete_stream()` removes the stream from the stream_list and calls `Http2Stream::initiating_close()`
- `Http2Stream::initiating_close()` set stream status closed, but sometimes doesn't call `Http2Stream::destroy()`

I bit worry about necessary write event signal to the HttpSM might be lost.